src/OrbitGl/PrimitiveAssembler.cpp

```diff
@@ -241,6 +244,29 @@ void PrimitiveAssembler::AddCircle(const Vec2& position, float radius, float z,
   }
 }
 
+void PrimitiveAssembler::AddVerticalArrow(float x, float start_y, float end_y, float z,
+                                          uint32_t width, float head_length_ratio,
+                                          float head_width_ratio, Color color) {
+  float min_y = std::min(start_y, end_y);
+  float max_y = std::max(start_y, end_y);
+
+  float head_length = std::abs(start_y - end_y) * head_length_ratio;
+  if (max_y == end_y) {
+    head_length = -head_length;
```

Comment:
Since you are just using `head_length + end_y` 4 times and not directly `head_length`, I will directly create a variable for that

Something like that (including the change I proposed to the .h file):
`float head_arrow_start_y = end_y + arrow_head_size[1] * (max_y == end_y ? -1 : 1);`